templates/config/controller/deployment.yaml.tpl

```diff
@@ -3,6 +3,7 @@ kind: Namespace
 metadata:
   labels:
     control-plane: controller
+    app.kubernetes.io/name: ack-system 
```

Comment:
I would leave this one out, since the Namespace isn't part of the "application" but rather all the parts of the "ack-system" application are Deployments.

Reply:
@jaypipes Does every ack controller install or attempt to install in the `ack-system` namespace?

Reply:
> Also, one other thing, @codekow: we have corresponding Helm templates in [`templates/helm`](https://github.com/aws-controllers-k8s/code-generator/tree/main/templates/helm) too.

I'll take a look at the helm charts. I can finish up this PR and associate another PR for the helm charts if it starts getting too complex.

Reply:
> @jaypipes Does every ack controller install or attempt to install in the `ack-system` namespace?

Yep!

Reply:
> > Also, one other thing, @codekow: we have corresponding Helm templates in [`templates/helm`](https://github.com/aws-controllers-k8s/code-generator/tree/main/templates/helm) too.
> 
> I'll take a look at the helm charts. I can finish up this PR and associate another PR for the helm charts if it starts getting too complex.

Cool. Let me know if I can help you in any way. Happy to assist! :)

Reply:
> > @jaypipes Does every ack controller install or attempt to install in the `ack-system` namespace?
> 
> Yep!

That is the reason I would consider the namespace as part of the `ack-system` since any controller has the namespace as a dependency?

@jaypipes What do you think?

Reply:
I still prefer to leave it off the Namespace. That said, I'm just one opinion... @A-Hilaly @acornett21 @RedbackThomson @jljaco what say you? :)

Reply:
> > @jaypipes Does every ack controller install or attempt to install in the `ack-system` namespace?
> 
> Yep!

@jaypipes This is not always the case, in OpenShift `ack-system` is the recommended namespace that a cluster admin should select, but the user can also choose the `openshift-operators` namespace from the drop-down in the UI as well. Since this is the case, `Namespace` should be excluded.

Reply:
I feel strongly that we should leave the annotation off the namespace. My three primary reasons:
1. Installing through Helm, we do not install it with a namespace (Helm takes care of the namespace) and therefore it will not have the annotation
2. The namespace is not exclusively for the ACK controllers. It's the default path for the controllers, but can be used by any other K8s objects, so it would be incorrect to label it as the ACK controller namespace
3. I have not found any other operators that annotate the namespace

Reply:
Thanks @RedbackThomson for chiming in here. :) That echoes my feelings as well.

Reply:
Agreed with @acornett21, @jaypipes  and @RedbackThomson here, ACK controllers can be installed in any namespace and can manage resources from any namespace, even the caches can watch resources outside of the `ack-system` namespace. https://github.com/aws-controllers-k8s/runtime/blob/main/pkg/runtime/cache/cache.go#L48-L59

Reply:
@RedbackThomson @A-Hilaly @jaypipes @acornett21 

Agree with all points. I work in OpenShift and everything you said aligns. Thanks for going through the discussion.